### PR TITLE
Add Docker

### LINKED ISSUE #2: Add Docker

### PR DESCRIPTION
**Pull request by pdany1116**

### CODE REVIEW

**Review by cazacutudor — CHANGES_REQUESTED** [review]
The app running steps are currently missing from the README, let's ensure we list them.

File: README.md [hunk 1]
```diff
@@ -1,2 +1,17 @@
...
+
+## Setup
+Step 1. Create development environment variables file and replace necessary variables 
+> cp .env.example .env
+
+Step 2. Build docker image
+> docker-compose build todo-api
+
+Step 3. Create development database
+> docker-compose run todo-api rails db:create db:migrate
```
Comment (cazacutudor) [inline README.md:12]: This step it's failing,

image

File: README.md [hunk 2]
```diff
@@ -1,2 +1,17 @@
...
+
+Step 4. Run todo-api
+> docker-compose up
```
Comment (cazacutudor) [inline README.md:15]: Also failing with,

image

File: .env.example [hunk 3]
```diff
@@ -0,0 +1,3 @@
+POSTGRES_USER=<postgres_user>
+POSTGRES_PASSWORD=<postgres_password>
+DATABASE_URL=postgres://${POSTGRES_USER}:${POSTGRES_PASSWORD}@postgres:5432/
```
Comment (cazacutudor) [inline .env.example]: Please specify the database name.

File: .env.example [hunk 4]
```diff
@@ -0,0 +1,3 @@
+POSTGRES_USER=<postgres_user>
+POSTGRES_PASSWORD=<postgres_password>
```
Comment (cazacutudor) [inline .env.example]: We should be able to get rid of user/password, for the local environment.

File: README.md (hunk 1 above)
Comment (pdany1116) [reply README.md:12]: 5739a6692b12025bee797b36ed35af721eefee4e

File: .env.example (hunk 3 above)
Comment (pdany1116) [reply .env.example]: 7692ee685afe3ee1c768dd6ea27d9f2729a00cc8

File: .env.example (hunk 4 above)
Comment (pdany1116) [reply .env.example]: 7692ee685afe3ee1c768dd6ea27d9f2729a00cc8

**Comment by pdany1116** [comment]
Build should work now. I've tested it on a fresh new system with clean docker environment.

**Review by cazacutudor — APPROVED** [review]

**Pull request merged**